File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Long,Integer>map = new HashMap<>();
        map.put(0L,1);
        int count=0;
        Long sum=0L;
        for( int num:nums){
            sum += num;
            if( map.containsKey(sum-k)){
                count+=map.get(sum-k);
            }
            if(!map.containsKey(sum)){
                map.put(sum,1);
            }
            else{
                int i=map.get(sum)+1;
                map.put(sum,i);
            }
        }
        return count;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        map={0:1}
        curr_sum=0
        for num in nums:
            curr_sum += num
            if curr_sum - k in map:
                count += map[curr_sum-k]
            map[curr_sum] = map.get(curr_sum,0) + 1
        return count